tests/conftest.py

```diff
@@ -1,14 +1,101 @@
 import os
-import tempfile
+import shutil
+from pathlib import Path
 
 import pytest
 
+TEST_ROOT = Path(__file__).resolve().parent
+
+
+def copy_migrations(migration, target_dir):
+    migration = TEST_ROOT / "migrations" / f"{migration}.py"
+    shutil.copy(migration, target_dir / "0.0.1.py")
+
 
 @pytest.fixture
-def tmpdir():
-    """Run the test in a temporary directory"""
+def tmp_migrations(tmp_path):
+    version_file = tmp_path / "version"
+    version_file.open("w").write("0.0.0")
     cwd = os.getcwd()
-    with tempfile.TemporaryDirectory(prefix="updater") as tmpdir:
-        os.chdir(tmpdir)
-        yield tmpdir
-        os.chdir(cwd)
+    os.chdir(tmp_path)
+    yield tmp_path
+    os.chdir(cwd)
+
+
+# Same as tmp_success_if_valid just to clarify from fixture names that a failing test may be
```

Comment:
So...I'm not totally sold on the value of these as fixtures. rollback, rollback_failure, cleanup, cleanup_failure, snapshot_failure are all only used once. They're all one line too.

 If `copy_migrations` was part of utils or something, the individual tests might be easier to read/understand?

Anyways, not requesting any change, just noting that I'm unsure about it.